### ISSUE TITLE
[no-unsafe-readonly-mutable-assignment] Doesn't catch unsafe assignment of function return types

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-plugin-total-functions#82]
```ts
type MutableA = { a: string };
type ReadonlyA = { readonly a: string };
type MutableFA = () => MutableA;
type ReadonlyFA = () => ReadonlyA;

const a: ReadonlyA = { a: "" };
const rofa: ReadonlyFA = () => a;
const mutfa: MutableFA = rofa; // not flagged; should be
```
